lib/versioning/VersionID.ts

```diff
@@ -269,6 +299,31 @@ export function base62Decode(str: string): string | Error {
 export const ENC_TYPE_HEX = 0; // legacy (large) encoding
 export const ENC_TYPE_BASE62 = 1; // new (tiny) encoding
 
+/**
+ * Checks if the given versionId string contains the specified format version.
+ *
+ * @param versionId - The versionId string to check.
+ * @param version - The expected format version.
+ * @returns true if the versionId contains the format marker and version, false otherwise.
+ */
+function hasVersionIDFormat(versionId: string, version: string): boolean {
+    // Format marker can only exist after the required versionId sections.
+    // This check removes the risk of looking for the format marker in the
+    // replication group ID, which can technically contain any character as
+    // it's set by the end user.
+    if (versionId.length <= LENGTH_TS + LENGTH_SEQ + LENGTH_RG + LENGTH_FT - 1) {
+        return false; // Not enough characters for format marker
+    }
+    // For constant time lookup, we always assume that the format marker is
+    // at the end of the versionId.
+    const formatMarkerIdx = versionId.length - LENGTH_FT;
+    const formatMarker = versionId.charAt(formatMarkerIdx);
+    if (formatMarker !== VersioningConstants.VersionId.FormatMarker) {
+        return false; // no format marker
+    }
```

Comment:
Nit: I'm not sure we need to create a variable for just one check

```suggestion
    if (versionId.charAt(formatMarkerIdx) !== VersioningConstants.VersionId.FormatMarker) {
        return false; // no format marker
    }
```